identity/client/src/main/java/com/azure/identity/IdentityClient.java

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.http.ProxyOptions.Type;
+import com.azure.core.implementation.serializer.SerializerAdapter;
+import com.azure.core.implementation.serializer.SerializerEncoding;
+import com.azure.core.implementation.serializer.jackson.JacksonAdapter;
+import com.azure.identity.implementation.MSIToken;
+import com.azure.identity.implementation.RefreshableTokenCache;
+import com.azure.identity.implementation.util.Adal4jUtil;
+import com.azure.identity.implementation.util.ScopeUtil;
+import com.microsoft.aad.adal4j.AsymmetricKeyCredential;
+import com.microsoft.aad.adal4j.AuthenticationContext;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.aad.adal4j.ClientCredential;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoSink;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.OffsetDateTime;
+import java.util.Random;
+import java.util.Scanner;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+/**
+ * The identity client that contains APIs to retrieve access tokens
+ * from various configurations.
+ */
+public final class IdentityClient {
+    private final IdentityClientOptions options;
+
+    /**
+     * Creates an IdentityClient with default options.
+     */
+    public IdentityClient() {
+        this.options = new IdentityClientOptions();
+    }
+
+    /**
+     * Creates an IdentityClient with the given options.
+     * @param options the options configuring the client.
+     */
+    public IdentityClient(IdentityClientOptions options) {
+        this.options = options;
+    }
+
+    /**
+     * @return the APIs to acquire tokens in Active Directory.
+     */
+    public ActiveDirectoryClient activeDirectory() {
```

Comment:
Because this is not a model class, we should not use fluent API design. I would call this method `getActiveDirectoryClient()`, same for `getManagedIdentityClient()` method below.

Reply:
Do you think this separation makes it more clear or more confusing? I can totally merge the methods in these 2 clients, putting them directly under IdentityClient.